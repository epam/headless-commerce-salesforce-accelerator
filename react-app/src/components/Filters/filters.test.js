/*
 * Copyright 2020 EPAM Systems, Inc. (https://www.epam.com/)
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import React from "react";
import { shallow } from "enzyme";

import BrandFilter from "./brandFilter";
import PriceFilter from "./priceFilter";
import ColorFilter from "./colorsFilter";
import SizeFilter from "./sizeFilter";

const defaultProps = {
  displayValue: "Brand",
  handleChange: () => null,
  selected: false,
  id: "brand",
};

describe("BrandFilter", () => {
  it("it should match snapshot", () => {
    const wrapper = shallow(<BrandFilter {...defaultProps} />);
    expect(wrapper).toMatchSnapshot();
  });
});

describe("PriceFilter", () => {
  it("it should match snapshot", () => {
    const wrapper = shallow(
      <PriceFilter {...defaultProps} displayValue="Price" id="price" />
    );

    expect(wrapper).toMatchSnapshot();
  });
});

describe("ColorFilter", () => {
  it("it should match snapshot", () => {
    const wrapper = shallow(
      <ColorFilter
        {...defaultProps}
        displayValue="Colors"
        id="refinementColor"
      />
    );

    expect(wrapper).toMatchSnapshot();
  });
});

describe("SizeFilter", () => {
  it("it should match snapshot", () => {
    const wrapper = shallow(
      <SizeFilter {...defaultProps} displayValue="Size" id="size" />
    );

    expect(wrapper).toMatchSnapshot();
  });
});
