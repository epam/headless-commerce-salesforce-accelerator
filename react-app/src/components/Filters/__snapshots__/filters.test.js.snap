// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BrandFilter it should match snapshot 1`] = `
<WithStyles(ForwardRef(FormControlLabel))
  control={
    <ForwardRef(WithStyles)
      checked={false}
      checkedIcon={
        <CheckboxCheckedIcon
          checked={true}
        />
      }
      icon={<CheckboxCheckedIcon />}
      onChange={[Function]}
    />
  }
  key="Brand"
  label={
    <ForwardRef(Styled(WithStyles(ForwardRef(Typography))))
      fontFamily="OpenSans"
      fontSize={14}
      fontWeight="normal"
    >
      Brand
    </ForwardRef(Styled(WithStyles(ForwardRef(Typography))))>
  }
/>
`;

exports[`ColorFilter it should match snapshot 1`] = `
<styled.div
  id="refinementColor"
  key="Colors"
  onClick={[Function]}
  selected={false}
>
  <styled.div
    backgroundColor="Colors"
    selected={false}
  />
</styled.div>
`;

exports[`PriceFilter it should match snapshot 1`] = `
<WithStyles(ForwardRef(FormControlLabel))
  control={
    <ForwardRef(WithStyles)
      checked={false}
      checkedIcon={
        <RadioButtonIcon
          checked={true}
        />
      }
      icon={<RadioButtonIcon />}
      onChange={[Function]}
    />
  }
  id="price"
  key="Price"
  label={
    <ForwardRef(Styled(WithStyles(ForwardRef(Typography))))
      fontFamily="OpenSans"
      fontSize={14}
      fontWeight="normal"
    >
      Price
    </ForwardRef(Styled(WithStyles(ForwardRef(Typography))))>
  }
  value="Price"
/>
`;

exports[`SizeFilter it should match snapshot 1`] = `
<styled.div
  id="size"
  key="Size"
  onClick={[Function]}
  selected={false}
  width={48}
>
  <Styled(Styled(WithStyles(ForwardRef(Typography))))
    selected={false}
  >
    Size
  </Styled(Styled(WithStyles(ForwardRef(Typography))))>
</styled.div>
`;
